compiler/qsc_hir/src/hir.rs

```diff
@@ -279,6 +279,57 @@ impl Display for Package {
     }
 }
 
+/// The name of a test callable, including its parent namespace.
+pub type TestCallableName = String;
+
+impl Package {
+    /// Returns a collection of the fully qualified names of any callables annotated with `@Test()`
+    pub fn get_test_callables(&self) -> Vec<(TestCallableName, Span)> {
+        let items_with_test_attribute = self
+            .items
+            .iter()
+            .filter(|(_, item)| item.attrs.iter().any(|attr| *attr == Attr::Test));
+
+        let callables = items_with_test_attribute
+            .filter(|(_, item)| matches!(item.kind, ItemKind::Callable(_)));
+
+        let callable_names = callables
+            .filter_map(|(_, item)| -> Option<_> {
+                if let ItemKind::Callable(callable) = &item.kind {
+                    if !callable.generics.is_empty()
+                        || callable.input.kind != PatKind::Tuple(vec![])
+                    {
+                        return None;
```

Comment:
So does this just silently drop a callable with the Test attribute if it happens to be generic? No warning or error?

Reply:
The error is detected in `qsc_passes`. Returning an error here doesn't show a pretty error to the user, so instead we detect it in passes and produce a spanned error. This pattern was suggested by @swernli and I think it is cleaner than returning an error here, which just prints an unspanned error in a VS Code dialog box.

Reply:
Yeah I saw those checks there. Just wondering the value of even testing for it here also, as the project is already in error from the other pass that detects these, and thus I assume shouldn't run anyway.

Reply:
It just prevents the test from showing up in the list. The pass doesn't filter out the callables from the HIR, because we prefer to preserve as much of the AST as possible for error recovery in passes. So even in an error state, the compilation _does_ return some HIR for a good user experience. We can't let those invalid callables show up in the test explorer right?